systemtest/src/main/java/io/strimzi/systemtest/resources/ResourceManager.java

```diff
@@ -272,72 +262,28 @@ private static void waitForDeletion(Kafka kafka) {
-        ReplicaSetUtils.waitForReplicaSetDeletion(KafkaMirrorMakerResources.deploymentName(kafkaConnect.getMetadata().getName()));
-
-        kubeClient().listPods().stream()
-                .filter(p -> p.getMetadata().getName().startsWith(KafkaConnectResources.deploymentName(kafkaConnect.getMetadata().getName())))
-                .forEach(p -> PodUtils.waitForPodDeletion(p.getMetadata().getName()));
-    }
-
-    private static void waitForDeletion(KafkaConnectS2I kafkaConnectS2I) {
-        LOGGER.info("Waiting when all the Pods are terminated for KafkaConnectS2I {}", kafkaConnectS2I.getMetadata().getName());
-
-        DeploymentUtils.waitForDeploymentConfigDeletion(KafkaMirrorMakerResources.deploymentName(kafkaConnectS2I.getMetadata().getName()));
-        ReplicaSetUtils.waitForReplicaSetDeletion(KafkaMirrorMakerResources.deploymentName(kafkaConnectS2I.getMetadata().getName()));
-
-        kubeClient().listPods().stream()
-                .filter(p -> p.getMetadata().getName().contains("-connect-"))
-                .forEach(p -> {
-                    LOGGER.debug("Deleting: {}", p.getMetadata().getName());
-                    kubeClient().deletePod(p);
-                });
-    }
-
-    private static void waitForDeletion(KafkaMirrorMaker kafkaMirrorMaker) {
-        LOGGER.info("Waiting when all the Pods are terminated for KafkaMirrorMaker {}", kafkaMirrorMaker.getMetadata().getName());
+    public static void waitForDeploymentDeletion(Deployment deployment) {
+        LOGGER.info("Waiting when all the pods are terminated for Deployment {}", deployment.getMetadata().getName());
 
-        DeploymentUtils.waitForDeploymentDeletion(KafkaMirrorMakerResources.deploymentName(kafkaMirrorMaker.getMetadata().getName()));
-        ReplicaSetUtils.waitForReplicaSetDeletion(KafkaMirrorMakerResources.deploymentName(kafkaMirrorMaker.getMetadata().getName()));
+        DeploymentUtils.waitForDeploymentDeletion(deployment.getMetadata().getName());
 
         kubeClient().listPods().stream()
-                .filter(p -> p.getMetadata().getName().startsWith(KafkaMirrorMaker2Resources.deploymentName(kafkaMirrorMaker.getMetadata().getName())))
+                .filter(p -> p.getMetadata().getName().startsWith(deployment.getMetadata().getName()))
                 .forEach(p -> PodUtils.waitForPodDeletion(p.getMetadata().getName()));
     }
 
-    private static void waitForDeletion(KafkaMirrorMaker2 kafkaMirrorMaker2) {
-        LOGGER.info("Waiting when all the Pods are terminated for KafkaMirrorMaker2 {}", kafkaMirrorMaker2.getMetadata().getName());
-
-        DeploymentUtils.waitForDeploymentDeletion(KafkaMirrorMakerResources.deploymentName(kafkaMirrorMaker2.getMetadata().getName()));
-        ReplicaSetUtils.waitForReplicaSetDeletion(KafkaMirrorMakerResources.deploymentName(kafkaMirrorMaker2.getMetadata().getName()));
+    public static <T extends HasMetadata> void waitForResourceDeletion(T resource, String deploymentName) {
```

Comment:
I don't see any benefit in this method. It's basically the same as `waitForDeploymentDeletion`